tools/scoring/score.py

```diff
@@ -1882,10 +1882,21 @@ def main():
             print(bl)
             print(lc)
 
-            if (g_args.bestof):
+            if (g_args.bestof != ""):
               # get winner *including non-competitive*
-              winners = grouped[(grouped["rank"] == 1)]
-              print("Best competitive solvers:\n" + winners[["name"]].to_csv())
+              winners = grouped[\
+                          (grouped["rank"] == 1) &\
+                          (grouped["score_correct"] > 0) &\
+                          (grouped["score_error"] == 0)
+                        ]
+              winners = winners.reset_index()\
+                .drop_duplicates(["year", "division"], keep="first")
```

Comment:
why drop_duplicates?

Reply:
Inside the rank == 1 the solvers are ordered by the other criteria of the scoring system (the CPU/wall-clock time mainly, at least in the single-query track). So the first of those solvers that has rank == 1 is the winner.